api/src/models.d.ts

```diff
@@ -62,3 +62,13 @@ export interface ParticipantActivities {
   principal_id: number;
   completed: boolean;
 }
+
+export interface ParticipantActivitiesWithCompletionStatus {
```

Comment:
This object structure defines a single activity and its completed status, so I think a better name for this would be `ParticipantActivityCompletionStatus`, switching to the singular form of activity.

Then, when you have multiple objects in an array it would read as `ParticipantActivityCompletionStatus[]`, which makes more sense since it's an array of singular objects.